src/routes/relay/entities/schema/sponsored-call.schema.ts

```diff
@@ -31,32 +29,15 @@ export const SponsoredCallSchema = z
     };
 
     if (isExecTransactionCall(data)) {
-      const isSafe = await isSafeContract(chainId, to);
-
-      // Non-Safe smart contract mimicing `execTransaction`
-      if (!isSafe) {
-        setError('Only `execTransaction` from Safes can be relayed');
-
-        return z.NEVER;
-      }
-
       return {
         ...values,
         safeAddress: to,
       };
     }
 
-    if (isMultiSendCall(data)) {
-      const isValid = await isValidMultiSendCall(chainId, to, data);
-
-      // MultiSend not containing only `execTransaction` calls from the same Safe
-      if (!isValid) {
-        setError('Invalid `multiSend` transaction');
-
-        return z.NEVER;
-      }
-
-      const safeAddress = await getSafeAddressFromMultiSend(chainId, data);
+    // MultiSend not containing only `execTransaction` calls
+    if (isValidMultiSendCall(chainId, to, data)) {
+      const safeAddress = getSafeAddressFromMultiSend(data);
```

Comment:
What if we deal with the negative case first:

```typescript
if (!isValidMultiSendCall(chainId, to, data)) {
  setError('Only (batched) Safe transactions can be relayed');
  return z.NEVER;
}

const safeAddress = getSafeAddressFromMultiSend(data);
if (!safeAddress) {
  setError('Cannot decode Safe address from `multiSend` transaction');
  return z.NEVER;
}

return {
  ...values,
  safeAddress,
};
```

In my opinion it looks a little bit clear that if a specific check fails we exit the function immediately (in the same way as before but we'd "read" that it'd do that after the if statement).

Reply:
Originally it was grouped according to transaction type: `execTransaction -> multiSend -> invalid` but I am a fan of early returns so I've adjusted it accordingly.